src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -161,8 +165,37 @@ private Expression CreateAssignmentFunc(Expression destinationFunc, bool constru
             actions.Add(_destination);
 
             return Block(actions);
-        }
-
+        }
+
+        private Expression HandlePath(PathMap pathMap)
+        {
+            var destination = ((MemberExpression)pathMap.DestinationExpression.ConvertReplaceParameters(_destination)).Expression;
+            var createInnerObjects = CreateInnerObjects(pathMap, destination);
+            var setFinalValue = CreatePropertyMapFunc(new PropertyMap(pathMap.DestinationMember, pathMap.TypeMap)
+            {
+                CustomExpression = pathMap.SourceExpression

```

Comment:
Can we handle this differently? The access modifier change is a bit odd. Maybe an interface or base class here?